src/Microsoft.DotNet.XHarness.Common/Logging/ILog.cs

```diff
@@ -13,15 +13,29 @@ public interface ILog : IDisposable
     {
         string Description { get; set; }
         bool Timestamp { get; set; }
-        string FullPath { get; }
         Encoding Encoding { get; }
-
         void Write(byte[] buffer, int offset, int count);
-        StreamReader GetReader();
         void Write(string value);
         void WriteLine(string value);
         void WriteLine(StringBuilder value);
         void WriteLine(string format, params object[] args);
         void Flush();
     }
+
+    /// <summary>
+    /// Interface to be implemented by those loggers that provide a StreamReader to read
+    /// the already added info.
+    /// </summary>
+    public interface IReadable
+    {
+        StreamReader GetReader();
```

Comment:
Why do we have this `IReadable`? Why not put `GetReader()` straight to `IReadableLog`?

Reply:
Seemed to be a good idea at the time, but know that you mention it, I'll remove that, lets see what happens, no the compiler helps :)

Reply:
There are classes that just care about this interface. Gives us freedom.